File: QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author david
 */
public abstract class Casilla {
    protected int coste;
    protected int numeroCasilla;    
    protected TipoCasilla tipo;
    
    public Casilla(int numeroCasilla, int coste, TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.tipo = tipo;
    }
    
    int getCoste() {
        return coste;
    }
    
    public int getNumeroCasilla() {
        return numeroCasilla;
    }
    
    public TipoCasilla getTipo() {
        return tipo;
    }
     
    
    abstract boolean soyEdificable();

    /*
    @Override
    public String toString () {
        return "Casilla{" + "\nNumero casilla=" + Integer.toString(numeroCasilla) + 
               ", Tipo de casilla=" + tipo +", Coste=" + Integer.toString(coste) + 
               ", Numero de hoteles=" + Integer.toString(numHoteles) + 
               ", Numero de casas=" + Integer.toString(numCasas) +
               "\nTitulo de propiedad=" + titulo + "}";
    }*/
}

File: QytetetRuby/lib/casilla.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#require_relative "titulo_propiedad"
#require_relative "tipo_casilla"

module ModeloQytetet
  class Casilla  
    attr_reader :coste, :tipo, :numero_casilla
    
    def initialize(num_casilla, coste, tipo)
      @numero_casilla = num_casilla
      @coste = coste
      @tipo = tipo
    end
    
    def soy_edificable
      return false;
    end
    
    def to_s
      return "\nCasilla \nNumero casilla: #{@numero_casilla}, Tipo de casilla: #{@tipo}"
    end
    
    private
#    def set_titulo_propiedad(nombre, factor_revalorizacion, alquiler_base,
#      hipoteca_base, precio_edificar)
#       if @tipo == TipoCasilla::CALLE
#        @titulo = TituloPropiedad.new(nombre, factor_revalorizacion, 
#        alquiler_base, hipoteca_base, precio_edificar)
#      end
#    end
    
#    def set_titulo_propiedad(titulo)
#      if @tipo == TipoCasilla::CALLE
#        @titulo = titulo
#        @titulo.casilla = self;
#      end
#    end
#    
  end
end